nuxt/index.js

```diff
@@ -1,39 +1,64 @@
 const { resolve } = require('path')
 
-function pickFirst(...args) {
-  for (const arg of args) {
-    if (arg !== undefined) {
-      return arg
+module.exports = function nuxtBootstrapVue(moduleOptions = {}) {
+  this.nuxt.hook('build:before', () => {
+    function pickFirst(...args) {
+      for (const arg of args) {
+        if (arg !== undefined) {
+          return arg
+        }
+      }
     }
-  }
-}
 
-module.exports = function nuxtBootstrapVue(moduleOptions = {}) {
-  // Merge moduleOptions with default
-  const options = {
-    ...this.options.bootstrapVue,
-    ...moduleOptions
-  }
-
-  const bootstrapVueCSS = pickFirst(options.bootstrapVueCSS, options.bootstrapVueCss, options.bvCSS)
-  if (bootstrapVueCSS) {
-    // Add BootstrapVue CSS
-    this.options.css.unshift('bootstrap-vue/dist/bootstrap-vue.css')
-  }
-
-  const bootstrapCSS = pickFirst(options.bootstrapCSS, options.bootstrapCss, options.css)
-  if (bootstrapCSS) {
-    // Add Bootstrap CSS
-    this.options.css.unshift('bootstrap/dist/css/bootstrap.css')
-  }
-
-  // Transpile src
-  this.options.build.transpile.push('bootstrap-vue/src')
-
-  // Register plugin, pasing options to plugin template
-  this.addPlugin({
-    src: resolve(__dirname, 'plugin.template.js'),
-    fileName: 'bootstrap-vue.js'
+    const kebabCase = str => str.replace(/([_\s]+([a-zA-Z])|([A-Z]))/g, (m, $1, $2, $3, o) => (o ? '-' : '') + ($2 || $3 || '').toLowerCase())
+    const pascalCase = str => str.replace(/(^|[-_\s]+)(.)/g, (m, $1, $2) => $2.toUpperCase())
+
+    // Merge moduleOptions with default
+    const options = {
+      ...this.options.bootstrapVue,
+      ...moduleOptions
+    }
+
+    const bootstrapVueCSS = pickFirst(options.bootstrapVueCSS, options.bootstrapVueCss, options.bvCSS)
+    if (bootstrapVueCSS) {
+      // Add BootstrapVue CSS
+      this.options.css.unshift('bootstrap-vue/dist/bootstrap-vue.css')
+    }
+
+    const bootstrapCSS = pickFirst(options.bootstrapCSS, options.bootstrapCss, options.css)
+    if (bootstrapCSS) {
+      // Add Bootstrap CSS
+      this.options.css.unshift('bootstrap/dist/css/bootstrap.css')
+    }
+
+    // Transpile src
+    this.options.build.transpile.push('bootstrap-vue/src')
```

Comment:
Is this still needed when we import everything from `bootstrap-vue/es` ?

Reply:
I'm think it is not needed, as it should be using the `es` files by default, and not `src`.  But we can check with @pi0 to be sure.

Reply:
Sometimes users may want to directly include a specific component. This just instructs nuxt to transpile if such pattern **seen**. So is safe to be here.